pages/common/views/components/form.jsx

```diff
@@ -1,34 +1,57 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import { map, get } from 'lodash';
 import { TextArea, Input, RadioGroup, Select } from 'govuk-react-components';
 import Snippet from '../containers/snippet';
+import ConditionalReveal from './conditional-reveal';
 
 const fields = {
   inputText: ({ ...props }) => <Input { ...props } />,
-  textarea: ({ value, ...props }) => <TextArea { ...props }>{ value }</TextArea>,
+  textarea: ({ value, ...props }) => <TextArea value={value} { ...props } />,
   radioGroup: ({ ...props }) => <RadioGroup { ...props } />,
   checkboxGroup: ({ ...props }) => <RadioGroup type="checkbox" { ...props } />,
-  select: ({ ...props }) => <Select { ...props } />
+  select: ({ ...props }) => <Select { ...props } />,
+  text: props => props.value &&
+    <Fragment>
+      <h3>{ props.label }</h3>
+      <p>{ props.value }</p>
+    </Fragment>
 };
 
 const Form = ({
   schema,
   model,
-  errors = {}
+  errors = {},
+  onFieldChange
 }) => (
   <form method="POST">
     {
-      map(schema, ({ inputType, accessor, ...props }, key) =>
-        fields[inputType]({
+      map(schema, ({ inputType, conditionalReveal, showIf, accessor, ...props }, key) => {
+        const field = fields[inputType]({
           key,
           value: accessor ? get(model[key], accessor) : (model[key] || ''),
           label: <Snippet>{`fields.${key}.label`}</Snippet>,
           hint: <Snippet optional>{`fields.${key}.hint`}</Snippet>,
           name: key,
           error: errors[key] && <Snippet>{`errors.${key}.${errors[key]}`}</Snippet>,
+          onChange: e => onFieldChange(key, e.target.value),
           ...props
-        })
-      )
+        });
+
+        if (showIf && !showIf(model)) {
+          return null;
+        }
+
+        return conditionalReveal
+          ? <ConditionalReveal
+            fieldName={key}
+            open={model[key]}
+            label={<Snippet>{`fields.${key}.conditionalReveal.label`}</Snippet>}
+            yesLabel={<Snippet>{`fields.${key}.conditionalReveal.yesLabel`}</Snippet>}
+            noLabel={<Snippet>{`fields.${key}.conditionalReveal.noLabel`}</Snippet>}
+          >{ field }</ConditionalReveal>
+          : field;
+      })
```

Comment:
I'd be way happier if this was another `if`. 10 lines seems a bit excessive for a ternary.

Reply:
I can only make one person happy a day, and today is looking like your lucky day. Please hold.

Reply:
#hashtagblessed